src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -355,4 +380,703 @@ String normalizeNativePath(String path) {
     String normalizeJavaPath(String path) {
         return path;
     }
+
+    //  Unix implementation of Files#copy and Files#move methods.
+
+    // minimum size of a temporary direct buffer
+    private static final int MIN_BUFFER_SIZE = 16384;
+
+    // calculate the least common multiple of two values;
+    // the parameters in general will be powers of two likely in the
+    // range [4096, 65536] so this algorithm is expected to converge
+    // when it is rarely called
+    private static long lcm(long x, long y) {
+        assert x > 0 && y > 0 : "Non-positive parameter";
+
+        long u = x;
+        long v = y;
+
+        while (u != v) {
+            if (u < v)
+                u += x;
+            else // u > v
+                v += y;
+        }
+
+        return u;
+    }
+
+    // calculate temporary direct buffer size
+    private static int temporaryBufferSize(UnixPath source, UnixPath target) {
+        int bufferSize = MIN_BUFFER_SIZE;
+        try {
+            long bss = UnixFileStoreAttributes.get(source).blockSize();
+            long bst = UnixFileStoreAttributes.get(target).blockSize();
+            if (bss > 0 && bst > 0) {
+                bufferSize = (int)(bss == bst ? bss : lcm(bss, bst));
+            }
+            if (bufferSize < MIN_BUFFER_SIZE) {
+                int factor = (MIN_BUFFER_SIZE + bufferSize - 1)/bufferSize;
+                bufferSize *= factor;
+            }
+        } catch (UnixException ignored) {
+        }
+        return bufferSize;
+    }
+
+    // whether direct copying is supported on this platform
+    private static volatile boolean directCopyNotSupported;
```

Comment:
Would it be possible to move MIN_BUFFER_SIZE and directCopyNotSupported to the top so they are declared before the instance fields?

Reply:
Fixed in b6cb94901d1bbaa34917a5aaf381ecf700282c54.